adapters/metax/metax.go

```diff
@@ -0,0 +1,244 @@
+package metax
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+	"text/template"
+
+	"github.com/prebid/openrtb/v20/openrtb2"
+	"github.com/prebid/prebid-server/v2/adapters"
+	"github.com/prebid/prebid-server/v2/config"
+	"github.com/prebid/prebid-server/v2/errortypes"
+	"github.com/prebid/prebid-server/v2/macros"
+	"github.com/prebid/prebid-server/v2/openrtb_ext"
+	"github.com/prebid/prebid-server/v2/util/ptrutil"
+)
+
+const SupportedCurrency = "USD"
+
+type adapter struct {
+	template *template.Template
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	errs := make([]error, 0, len(request.Imp))
+
+	if len(request.Imp) == 0 {
+		return nil, []error{&errortypes.BadInput{
+			Message: "No impressions in the request",
+		}}
+	}
+
+	// split impressions
+	reqDatas := make([]*adapters.RequestData, 0, len(request.Imp))
+	for i := range request.Imp {
+		imp := &request.Imp[i]
+		impCopy := *imp
+		requestCopy := *request
+
+		metaxExt, err := parseBidderExt(imp)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		if err := validateParams(metaxExt); err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		if err := preprocessImp(&impCopy); err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		endpoint, err := a.getEndpoint(metaxExt)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		impCopy.Ext = nil
```

Comment:
@metax-kehan any particular reason to set Imp Ext to nil ?

Reply:
no particular reason, we can keep the ext

Reply:
no particular reason here, we can keep it, thanks for your reminder